src/test/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingServiceTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.companieshouse.overseasentitiesapi.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+import uk.gov.companieshouse.api.ApiClient;
+import uk.gov.companieshouse.api.model.filinggenerator.FilingApi;
+import uk.gov.companieshouse.api.model.transaction.Transaction;
+import uk.gov.companieshouse.overseasentitiesapi.client.ApiClientService;
+import uk.gov.companieshouse.overseasentitiesapi.exception.ServiceException;
+import uk.gov.companieshouse.overseasentitiesapi.exception.SubmissionNotFoundException;
+import uk.gov.companieshouse.overseasentitiesapi.mocks.Mocks;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.when;
+import static uk.gov.companieshouse.overseasentitiesapi.utils.Constants.FILING_KIND_OVERSEAS_ENTITY;
+
+@ExtendWith(MockitoExtension.class)
+public class FilingServiceTest {
+
+    private static final String OVERSEAS_ENTITY_ID = "abc123";
+    public static final String FILING_DESCRIPTION = "Test this";
+
+    @InjectMocks
+    private FilingsService filingsService;
+
+    @Mock
+    private OverseasEntitiesService overseasEntitiesService;
+
+    @Mock
+    private ApiClientService apiClientService;
+
+    @Mock
+    private ApiClient apiClient;
+
+    private Transaction transaction;
+
+    @Test
+    void test() throws SubmissionNotFoundException, ServiceException {
```

Comment:
Suggest changing this method name to explain what the method is testing. So something like 'testSuccessfulFilingGeneration'.